File: app/src/main/java/ca/danielw/rankr/activities/IntroActivity.java
package ca.danielw.rankr.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import ca.danielw.rankr.R;

public class IntroActivity extends AppCompatActivity {

    private FirebaseAuth mAuth;

    private Button btnSignIn;
    private Button btnCreateLeague;
    private Button btnSignUp;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.intro_1);

        mAuth = FirebaseAuth.getInstance();

        FirebaseUser currentUser = mAuth.getCurrentUser();

        btnSignIn = (Button) findViewById(R.id.btnSignIn);
        btnCreateLeague = (Button) findViewById(R.id.btnCreateLeague);
        btnSignUp = (Button) findViewById(R.id.btnSignUp);

        if (currentUser != null) {
            //Open to leaderboard
//            Intent intent = new Intent(this, MainActivity.class);
//            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
//            startActivity(intent);
        }

        btnSignIn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(IntroActivity.this, SignInActivity.class);
                startActivity(intent);
            }
        });

        btnCreateLeague.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(IntroActivity.this, CreateLeagueActivity.class);
                startActivity(intent);
            }
        });

        btnSignUp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(IntroActivity.this, SignUpActivity.class);
                startActivity(intent);
            }
        });

    }

}

File: app/src/main/java/ca/danielw/rankr/IntroActivity.kt
package ca.danielw.rankr

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import kotlinx.android.synthetic.main.intro_1.*
import android.content.Intent
import ca.danielw.rankr.activities.CreateLeagueActivity
import ca.danielw.rankr.activities.SignInActivity


class IntroActivity : AppCompatActivity() {

    private val TAG = IntroActivity::class.java.name

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.intro_1)

        btnSignIn.setOnClickListener {
            Log.e(TAG, "Sign in")
            //launch the signin flow

            val myIntent = Intent(this@IntroActivity, SignInActivity::class.java)
            this@IntroActivity.startActivity(myIntent)
        }

        btnCreateLeague.setOnClickListener {
            Log.e(TAG, "Create league")
            //Start create league flow

            val myIntent = Intent(this@IntroActivity, CreateLeagueActivity::class.java)
            this@IntroActivity.startActivity(myIntent)

        }
    }

}
